lib/components/tabs/tabs.dart

```diff
@@ -0,0 +1,136 @@
+import 'dart:html';
+import 'dart:async';
+import 'package:katikati_ui_lib/components/logger.dart';
+
+var logger = Logger('Tabs');
+
+class TabView {
+  String id;
+  String label;
+  DivElement content;
+
+  TabView(this.id, this.label, this.content);
+}
+
+class TabsView {
+  DivElement renderElement;
+  List<TabView> _tabs;
+  Map<String, TabView> _tabsLookup;
+
+  String _selectedTabID;
+  String _tabContentClassname;
+
+  List<SpanElement> _tabChoosers = [];
+  DivElement _tabsHeader;
+  DivElement _tabContent;
+
+  Stream<String> _onEmpty;
+  StreamController<String> _onEmptyController;
+  Stream<String> get onEmpty => _onEmpty;
```

Comment:
I don't think we want to use this mechanism for managing an empty set of tags. What I suggest we do is instead make the `TabsView` be able to smoothly handle the case with no tabs, and just display a "no tabs available" message (this text could actually be configurable).

Feel free to pick this up in a separate PR though rather than resolve here if that's easiest!